examples/onchain-signer/hardhat.config.ts

```diff
@@ -26,7 +26,17 @@ const config: HardhatUserConfig = {
     'sapphire-testnet': { ...sapphireTestnet, accounts },
     'sapphire-localnet': { ...sapphireLocalnet, accounts },
   },
-  solidity: '0.8.20',
+  solidity: {
+    version: '0.8.20',
+    settings: {
+      // XXX: Needs to match https://github.com/oasisprotocol/sapphire-paratime/blob/main/contracts/hardhat.config.ts
```

Comment:
Based on some of my quick testing I believe it is not about matching, but actually things fail without this optimization enabled.

For example, removing this optimization in contracts/hardhat.config.ts makes the tests in contracts/ fail as well.

Do we know why this happens? Should someone have investigate this further?